lib/src/kvs/tx.rs

```diff
@@ -649,50 +674,71 @@ impl Transaction {
 	///
 	/// This function fetches the full range of key-value pairs, in a single request to the underlying datastore.
 	#[allow(unused_variables)]
-	pub async fn scan<K>(&mut self, rng: Range<K>, limit: u32) -> Result<Vec<(Key, Val)>, Error>
+	pub async fn scan<K>(
+		&mut self,
+		page: ScanPage<K>,
+		batch_limit: u32,
+	) -> Result<ScanResult<K>, Error>
 	where
-		K: Into<Key> + Debug + Clone,
+		K: Into<Key> + From<Vec<u8>> + Debug + Clone,
```

Comment:
The `From<Vec<u8>>` shouldn't be needed here, because of the `Into<Key>` which is the same thing?

Reply:
We actually need From as well, because we need to have a way of recreating Key from Vec<u8>